test/e2e/Apps/BasicPowerShell/RaiseComplexException/run.ps1

```diff
@@ -0,0 +1,17 @@
+#
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+param($InstanceId)
+
+Import-Module -Name "MyHelperModule"
+
+if (GetExecutionCount($InstanceId) -gt 0) {
```

Comment:
I will strongly encourage you to avoid this usage of parentheses with PowerShell, as it looks deceptively similar to other languages and invites incorrect usage. For example, if someone ever decide to add one more parameter, this code will lead to _surprising_ results: `GetExecutionCount($InstanceId, $somethingElse)`.

Consider rewriting it like this:

```powershell
if ((GetExecutionCount $InstanceId) -gt 0) {
    IncrementExecutionCount $InstanceId
    "Success"
}
else {
    IncrementExecutionCount $InstanceId
    throw [System.InvalidOperationException]::new("This activity failed\r\nMore information about the failure", [System.OverflowException]::new("Inner exception message"))
}
```

Reply:
Fixed